### ISSUE TITLE
feat: improve system info section for improved understanding

### ISSUE THREAD

**Issue by mathetos** [impress-org/givewp#3932]
# Feature Request

## User Story

As a Give Support Technician or Developer, or anyone troubleshooting a site that Give is on, I want to have actionable information on the System Info panel. Currently, there is a lot of information on this page, and there are some status indicators, but some information on this page is really crucial, while a lot of it is only tangentially relevant or barely relevant at all.

It would be most useful for Support Techs and Devs to have the very first information from this panel be focused purely on the must-have Give Requirements.

## Visuals

Here's my proposed minimum requirements section. This is up for discussion of course, but these items are required by Give in order to function and without them things really do go in unexpected ways. 

![image](https://user-images.githubusercontent.com/930724/50936267-dc10a500-1423-11e9-9ed0-9f499447365e.png)

## Possible Solution

1. Add a new section as the FIRST section of the System Info
2. Add a link directly above or below this section with a link to our Give Requirements Doc with this shortlink: `http://docs.givewp.com/requirements`
3. Additionally, if there are any "red" items, show an Admin notice throughout the site that the Give System Info needs attention. When on that page, there should be an alert above the System Info with a note about the Requirements with a link to our Requirements page. That Doc will be updated to have tips about updating their requirements to give users actionable information to resolve the problem.

## Related

* See our [Slack conversation](https://givewp.slack.com/archives/C0E290T6Z/p1547051627080100) about a similar issue with Database updates and where this idea was discussed.

## Acceptance Criteria

- [ ] The "Give Requirements" section is added and each individual item is removed from the other sections.
- [ ] When the System Info is copied, this information gets copied correctly.
- [ ] When a Requirement is not met, an Admin notice is displayed (for Admin users only) in the backend alerting them that the Give System Info needs attention.
- [ ] When that alert appears, a message is displayed above the System Info describing what the problem is exactly with a link to our Give Requirements Doc
- [ ] The Give Requirements Doc is updated to have tips to help users understand how they can update their requirements to meet our minimums (a Give-Docs issue will be made for this).

**Comment by kevinwhoffman** [comment]
Before doing any major refactoring, we should discuss this in light of the Health Check & Troubleshooting plugin being merged into WordPress core, possibly as soon as 5.2 per [this Trac ticket](https://core.trac.wordpress.org/ticket/39165).

The System Info screen in WP Business Reviews was developed based on the Health Check code base so that it can extend the Health Check plugin's report once it merges into core. Here's the WPBR pull request for reference: https://github.com/impress-org/wp-business-reviews/pull/286

@mathetos Do you still foresee Give having its own System Info screen even after Health Check merges into core, or do you see Give extending the Health Check report with its own section?

**Comment by mathetos** [comment]
I should have mentioned that. Perhaps I'm putting too many different things into this one issue, and perhaps YES it's best to wait until HealthCheck is merged into Core. BUT... some of my desires here are still independent from that issue. Specifically:

1. When customers provide us System Info (which honestly, they do A LOT -- I'm surprised but they provide it very regularly), the very first top information should be about the Give Requirements. I'm not convinced we can do that reliably with the Core System Info.

2. I feel strongly that users need to be very specifically informed if they are running a stack that just simply won't work well with Give. We bend over backwards to be as backwards compatible as possible, but if their cURL (for example) is 15+ years old, a notice informing them of that in advance would help set expectations in Support a lot and perhaps help them be better informed of their crappy hosting.

3. I also forgot to mention the improved hosting check that we added into WPBR. I am hoping we can use that to have more accurate info on WHO the webhosts are of these customers. 

So there's several layers of issues here, but I think we can resolve them with one concerted effort. 

Bottom line though: We don't have any real guarantees of what the Core SysInfo will look like, or how flexible the hooks will be or anything, so i think it's best to rely on our own instead of wait and then find out we might have a LESS optimal experience leaning on that.

**Comment by kevinwhoffman** [comment]
Closing due to inactivity. Also, the Tools > System Info screen has been iterated upon since this issue was created. For example, the web host is now displayed.
